### ISSUE TITLE
Quizzes and Polls not completing (or even attempting)

### ISSUE THREAD

**Issue by ChippewaBarr** [h4x0rm1k3/ABS#3]
Seems to be a bug with some of the quizzes and polls.

I usually open Chrome and leave it to do it's thing (since sometimes it takes a long time for the rewards page to 'realize' that the task has been completed) but it seems it's having trouble completing (or even loading) some quizzes.

The tabs will open but the quizzes/polls will not complete, and even further, the quizzes and polls won't even load at the bottom of the page as per normal...the page just contains the search results.

I've uninstalled and reinstalled the extension, reset Chrome and tried Incognito mode, but no change. 

Daily streak search and other searches work fine, but quizzes and polls seem to be having issues,

**Comment by ChippewaBarr** [comment]
Here's what I get when these specific quizzes open: (https://imgur.com/a/nKdHqhw)

And here's a paste of the generated URLs (https://pastebin.com/ZrRnAw0r)

**Comment by wearysky** [comment]
I've noticed similar weird behaviour - it opens the links to everything but doesn't seem to complete them. I'm wondering if it's a timing issue? I might try messing around with the source a little bit, and put a delay in between every link's "click" and see if that makes a difference.

**Comment by wearysky** [comment]
So, just an update: 30 seconds didn't appear to be long enough, but increasing the delay between each "click" to 60 seconds seems to have fixed the issue. Now when I wake up in the morning, all of my polls and quizzes work.

**Comment by wearysky** [comment]
@ChippewaBarr I forked and added my change (that's the only difference) here: https://github.com/wearysky/ABS

**Comment by h4x0rm1k3** [comment]
For some reason i'm not getting notifications and only just kind of worked out what was happening this morning and was thinking of finding where to add a wait or timer before the rewards started because it seems as soon as search runs it takes over so my thinking was to try and add a 3 minute delay to rewards starting which should be plenty long enough for searches to complete and allow rewards to also complete after the 3 minute delay. I'm glad @wearysky jumped in and found a similar solution to what I was thinking so if that works then it's worth sticking with that and i'll fork it from there if ok to keep this git up to date as well. If by any chance it breaks again though similarly or reports that it doesn't work come in then it might be worth looking in to adding the 3 minute delay to rewards starting so that searches complete before rewards start going.

**Comment by h4x0rm1k3** [comment]
I see now that the fix by @wearysky does exactly what I was intending on trying to implement myself in that both the searches and initial rewards tabs open but only the searches run (even if you try to click the open rewards tasks button in the ABS add on menu it still has that timer), then a minute later after the searches have easily completed the rewards tabs will open up, perfect! I will say though that I noticed a few of my searches weren't being completed a while back so I upped the amount of searches my accounts do each day so i altered my own personal code I run to use 3 minutes instead of 1, the code I changed in my Git is the 1 minute version. Thanks again for the fix @wearysky

**Comment by ChippewaBarr** [comment]
Good work @wearysky @h4x0rm1k3 for keeping this going. @h4x0rm1k3 is this new 3 minute delay present in your latest release? If so, I'll download and reinstall again.

And If I'm following this correctly...what it does is lets the searches take place over three minutes, and when that three minutes is up, it begins the quizzes and polls?

**Comment by h4x0rm1k3** [comment]
@ChippewaBarr My code only has the 1 minute delay and not 3 minutes, my personal use code has a 3 minute delay because I do a lot more searches so 1 minute just won't do. As for how it runs you've got it exactly right, searches 1st for 1 minute and then rewards kick in after that delay.

**Comment by wearysky** [comment]
Looks like whatever was going on with today's tasks, my changes didn't work. @h4x0rm1k3 did your local changes work better for you today?

**Comment by wearysky** [comment]
And to be clear, I'm totally fine with my changes going away, I only created the fork just in case this one had been abandoned :)

**Comment by h4x0rm1k3** [comment]
@wearysky Yes for the past 3 days now mine have ran perfectly with the delay, i've just checked in on todays ones and they're running along nicely as well so my guess is that your searches took a little longer than usual today so it might be worth rising your delay a little further to see if that helps at all. If I get a few more reports of them still not completing with a 1 minute delay i'll update the code to what I use myself in the hopes that it works for all that way. No probs forking and updating the code, it's what I did myself anyway as the original guy wasn't updating it anymore.

**Comment by wearysky** [comment]
Failed again today. Gonna update the delay to 3 minutes and see how it goes. It's running at 2am so it's not like it really matters what the delay is :)

**Comment by ChippewaBarr** [comment]
Yeah mine's not working as well either.

@h4x0rm1k3 would you be able to update the code to include the 3min delay? I'm not exactly the best at coding this type of stuff lol.

**Comment by h4x0rm1k3** [comment]
@ChippewaBarr Done, the code now has a 3 minute delay before rewards start. That should be long enough for around 100-110 searches total as I do 50 desktop and 40 mobile searches and that takes 2 minutes and 30 seconds when I run mine.

**Comment by ChippewaBarr** [comment]
@h4x0rm1k3 awesome thanks for the change!

I'll run your new version for a while and see if anything goes wrong lol, here's hoping!

**Comment by wearysky** [comment]
So, I pulled the latest yesterday, and the "daily set" tasks are all completing with the latest code, but not the "more activities" ones.  Odd. I hit "open reward tasks" again and it completed correctly. Not a huge deal, tbh, as long as the daily set runs the streak stays alive, and that's the most important part. :)

**Comment by h4x0rm1k3** [comment]
@wearysky Very strange that  is then 'cos my 'more activities' get done no problem now, in fact, it's back to how it used to work for me again now albeit with a 3 minute delay on rewards tasks so not sure what's going on your end. I seen your message on the manifest issues post so i'll look in to that shortly and see about updating the file.

**Comment by wearysky** [comment]
@h4x0rm1k3 yeah, when it was failing before, the windows would be open they just weren't registered as being clicked. I'm gonna throw some console logging in to see if it's actually finding the links to "click" or not. If not, then something real weird is going on.

**Comment by h4x0rm1k3** [comment]
What I found happening before was that all searches & rewards tabs would open and work alongside each other and then at some point it changed in that as soon as searches ran then every window would be taken over by the searches and only the quick 10 pointers was being done during the period when the tabs open and the 1st search completes which is enough for 'most' of the 10 pointers apart from polls to be done and polls and the 30 pointer ones weren't.

What i've found now with the 3 minute delay is that the searches will start straight away and the initial rewards tab will open but the rewards tasks themselves won't start opening up their tabs and doing their thing for 3 minutes in which time searches are usually done so can't interfere with the rewards tasks running which seems to work perfectly for me, or at least has been since I implemented the 3 minute delay and updated the code.

Strange what's happening with yours, have you tried removing ABS completely and starting from scratch, see if that helps at all?

**Comment by wearysky** [comment]
@h4x0rm1k3 yeah, every time I make a change to the code, I remove ABS completely, and re-install, and re-set my options. I'm actually going to add in another delay before it even searches for the cards, see if that improves things

**Comment by wearysky** [comment]
Monday morning update: still not clicking the "More activities" in the bottom. I added in a delay before it even looks at the cards, still no dice - it finds the three main cards at the top, but when it loops through all the cards it isn't seeing the mee-icon-AddMedium class on the "More Activies" ones. But when I wake up in the morning and manually run the code in the console to fetch the cards and loop through them on the window that's still sitting there, open, it *does* see the cards. So very bizarre. I'm gonna try on Firefox, see if that makes a difference, maybe there's something weird going on with my CHrome.

**Comment by h4x0rm1k3** [comment]
Try this, it's just the src folder zipped up so easy enough to deal with - https://www.mediafire.com/file/cu4ihr0dc1usj00/src.zip/file

**Comment by wearysky** [comment]
@h4x0rm1k3 thanks, I'll give it a try! I did a quick compare and yeah, don't see much diff there... But we'll see how it goes tmw!

**Comment by ChippewaBarr** [comment]
@wearysky @h4x0rm1k3 just as an FYI, mine has been running correctly since the update with the 3min delay. So I'll run with this version until it (hopefully doesn't) mess up again!

@wearysky do you keep Chrome open all the time? I found previously after a few days of it being open without being closed, is when problems start to arise. I can only assume it's on Chrome's end somewhere. My Chrome is pretty much stock settings with only this extension and Ublock Origin running on it.

**Comment by wearysky** [comment]
@h4x0rm1k3 happened to me again today even with your code, so it's clearly not a code issue. I do often leave Chrome open, but I rebooted a couple times yesterday so it's not like it had been open for days. So weird! I'm gonna keep dumping different logging info till I figure out exactly what's going on.

**Comment by h4x0rm1k3** [comment]
@wearysky have you tried chromium as i'm sure that sets up in it's own folders separate to chrome and should run ABS fine still or you could even set up a quick VM and set it to run an account on a fresh OS install with just chrome installed along with the extension & account details etc.

**Comment by wearysky** [comment]
@h4x0rm1k3 I'd rather not bother with a VM... But yeah, you can run Chrome and Chrome Canary side by side on Windows so maybe I'll try that and see how it goes.

**Comment by wearysky** [comment]
Fascinating: absolutely clean Chrome Canary install, literally all I did was install the extension and login to my Microsoft account... And still the same behaviour. Guess it's time to try a VM with a fresh OS install and see how that behaves.

**Comment by h4x0rm1k3** [comment]
@wearysky definitely a strange 1 and judging by other comments and no other reports it seems isolated to you for some reason! With any luck knocking up a quick freshly installed OS on VM with just Chrome and your account will help in at least letting you know if it's something on your computer somewhere. Maybe completely deleting the Chrome/userdata folder in appdata/local might help as there may be remnants of your old setup left behind, in fact go through all of appdata, progdata & program files/(x86) folders and delete all Chrome folders in any Google folders and with any luck that might just delete the problem along with it!

**Comment by wearysky** [comment]
So, just a final follow-up - I changed my start time from 2am to 4am, thinking maybe for some reason the "more activities" hadn't reset properly for me yet by 2am (console logging was showing that there were only the main 3 "daily set" activity cards with the Add class on them)... And so far so good. It looks like it's been working. Gonna keep an eye on it the next few days to see if there's any more hiccups.

**Comment by wearysky** [comment]
@h4x0rm1k3 just as a final follow up, it's been smooth sailing all week. Must have been the time of day it was running.

**Comment by wearysky** [comment]
@h4x0rm1k3 not sure if you've had this issue recently, but I noticed a couple times over the last few weeks that rewards quizzes and polls sometimes aren't appearing, even when manually clicked. I've got a couple VMs set up with separate accounts that literally just have Windows and Chrome with the plugin installed on them , nothing else, and the plugin clicks the links but there's no quiz or poll on the new tab. Just the search results and that's it. Only one error in the dev console (an uncaught reference error, "define is not defined"), but that error shows up on the other ones that work on the same day, too. Very strange! Wiped out all my cache, cleared out everything, still happens. The next day it's fine again. Not a *massive* deal since it only happens every once in a while, but annoying because it wipes out the streak so you lose the bonus points and have to start over again.

**Comment by wearysky** [comment]
Meant to add: it seems completely unrelated to the plugin, as even with the plugin fully removed from Chrome, the poll and quiz still don't show up. Even trying in different browsers (Edge, Firefox), still no luck. It's just an odd thing I noticed and was wondering if anybody else had seen it and might have a solution.

**Comment by h4x0rm1k3** [comment]
@wearysky I think I may have noticed something like that once where a couple of my accounts 2 of the polls & quizzes weren't ticked, I can't remember if the tabs had been opened or not but I do know I could manually open and complete them that way. It's very possible it's something completely unrelated as the majority of my accounts ran fine and those that had issues worked fine after and have done ever since and has literally only been the 1 time. Other than that I have no idea, I just checked on some of mine that I have running at this time and they all seem to be working perfectly still and I haven't noticed any other reports of issues at all so at the moment I can't help until it affects me or someone else to try and fix, sorry.

**Comment by wearysky** [comment]
@h4x0rm1k3 I'm not 100% sure why it's happening, but I found a fix this morning. It happened again with my main account this morning, and with a little more experimentation, it's either Microsoft blocking my IP (which seems unlikely, as all of the other rewards tasks still work), or something broken with routing with my ISP. I connected to a VPN in the same country as me, and boom, polls and quizzes pop up again.
